mm2src/coins/coin_errors.rs

```diff
@@ -90,6 +91,38 @@ impl From<keys::Error> for ValidatePaymentError {
     fn from(err: keys::Error) -> Self { Self::InternalError(err.to_string()) }
 }
 
+impl From<NftAssocTypesError> for ValidatePaymentError {
+    fn from(err: NftAssocTypesError) -> Self { Self::InternalError(err.to_string()) }
+}
+
+impl From<CoinAssocTypesError> for ValidatePaymentError {
+    fn from(err: CoinAssocTypesError) -> Self { Self::InternalError(err.to_string()) }
+}
+
```

Comment:
You can use error_enum macro for this conversion so we don't need to write it

Reply:
Sorry could you clarify what macro do you mean, just can find such
![image](https://github.com/KomodoPlatform/komodo-defi-framework/assets/52405288/733c758a-1fc4-4b00-a14c-32b89c193965)

Reply:
sorry I mean `EnumFromStringify`...e.g instead of writing these
```rust
impl From<rlp::DecoderError> for ValidatePaymentError {
    fn from(err: rlp::DecoderError) -> Self { Self::TxDeserializationError(err.to_string()) }
}

impl From<serialization::Error> for ValidatePaymentError {
    fn from(err: serialization::Error) -> Self { Self::TxDeserializationError(err.to_string()) }
}
```
you can just do this
```rust
#[derive(Debug, Display, EnumFromStringify)]
pub enum ValidatePaymentError {
    /// Should be used to indicate internal MM2 state problems (e.g., DB errors, etc.).
    InternalError(String),
    /// Problem with deserializing the transaction, or one of the transaction parts is invalid.
    #[from_stringify("serialization::Error", "rlp::DecoderError")]
    TxDeserializationError(String),
    ...
}
```

Please consider replacing similar code in this mod if you have enough time as it's going to reduce the codes by some margin

Reply:
completely forgot about this macro, thanks! changed where noticed